content/tutorial-deep-reinforcement-learning-with-pong-from-pixels.md

```diff
@@ -0,0 +1,820 @@
+- The agent can choose an action based on cumulative _rewards_ (also known as the _value function_) and the _policy_. The _cumulative reward function_ estimates the quality of the observations the agent visits using its _policy_.
+- The _policy_ (defined by a neural network) outputs action choices (as (log) probabilities) that should maximize the cumulative rewards from the state the agent is in.
+- The _expected return from an observation_, conditional to the action, is called the _action-value_ function. To provide more weight to shorter-term rewards versus the longer-term ones, you usually use a _discount factor_ (often a floating point number between 0.9 and 0.99).
+- The sequence of actions and states (observations) during each policy "run" by the agent is sometimes referred to as a _trajectory_ — such a sequence yields _rewards_.
+
+You will train your Pong agent through an "on-policy" method using policy gradients — it's an algorithm belonging to a family of _policy-based_ methods. Policy gradient methods typically update the parameters of the policy with respect to the long-term cumulative reward using [_gradient descent_](https://en.wikipedia.org/wiki/Gradient_descent) that is widely used in machine learning. And, since the goal is to maximize the function (the rewards), not minimize it, the process is also called _gradient ascent_. In other words, you use a policy for the agent to take actions and the objective is to maximize the rewards, which you do by computing the gradients and use them to update the parameters in the policy (neural) network.
+
++++ {"id": "6sWBMi61EOok"}
+
+## Set up Pong
+
++++ {"id": "k7UEY24ACQVp"}
+
+1. First, you should install OpenAI Gym (using `pip install gym[atari]` - this package is currently not available on conda), and import NumPy, Gym and the necessary modules:
+
+```{code-cell} ipython3
+:id: 7cWZsoNtKbFy
+
+import numpy as np
+import gym
+```
+
++++ {"id": "zkku9ZIdRfvm"}
+
+Gym can monitor and save the output using the `Monitor` wrapper:
+
+```{code-cell} ipython3
+:id: OKKaJHmRRfvm
+
+from gym import wrappers
+from gym.wrappers import Monitor
+```
+
++++ {"id": "wk69Jer36elU"}
+
+2. Instantiate a Gym environment for the game of Pong:
+
+```{code-cell} ipython3
+:id: m_2E9b6j5bkp
+
+env = gym.make("Pong-v0")
+```
+
++++ {"id": "J0M7me6d5mrK"}
+
+3. Let's review which actions are available in the `Pong-v0` environment:
+
+```{code-cell} ipython3
+:id: 6zEAzKEu5cU0
+
+print(env.action_space)
+```
+
+```{code-cell} ipython3
+:id: R_kO52Hr5_5b
+
+print(env.get_action_meanings())
+```
+
++++ {"id": "SbBxzJ0l6HDz"}
+
+There are 6 actions. However, `LEFTFIRE` is actually `LEFT`, `RIGHTFIRE` — `RIGHT`, and `NOOP` — `FIRE`.
+
+For simplicity, your policy network will have one output — a (log) probability for "moving up" (indexed at `2` or `RIGHT`). The other available action will be indexed at 3 ("mode down" or `LEFT`).
```

Comment:
@melissawm @rossbar I spotted a tiny typo here 🤦 shall we fix it here or after the PR is merged?

```suggestion
For simplicity, your policy network will have one output — a (log) probability for "moving up" (indexed at `2` or `RIGHT`). The other available action will be indexed at 3 ("move down" or `LEFT`).
```